federation/schema_syncer.go

```diff
@@ -0,0 +1,79 @@
+package federation
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/samsarahq/go/oops"
+	"github.com/samsarahq/thunder/graphql/introspection"
+)
+
+// SchemaSyncer has a function that checks if the schema has changed,
+// and if so updates the planner in the federated executor
+type SchemaSyncer interface {
+	UpdatePlanner(ctx context.Context, optionalArgs interface{}) (*Planner, error)
```

Comment:
I think we need to create an external NewPlanner function?

Reply:
I renamed it to `FetchPlanner` since we also use this function to create a new planner. Let me know if this sounds okay!